### DISCUSSION TITLE
Appreciation and Thanks

### DISCUSSION THREAD

**Discussion by tsktech** [make-all/tuya-local#485] [General]
Dear Jason,

Thanks for the "Tuya-Local", works out o the box, 
been with HA for almost 6 years, could never get the Tuya (smart life) to HA. this integration is just WOW!!

Regards
Srikanth Kamath, Bangalore, India

**Comment by yarafie** [comment]
Would like to second the above appreciation.
Although The default HA Tuya integration worked for me i wsnted to bring things local and out of the cloud.

Just few questions though.
1. I assume I have to first pair any wifi device first with the Smart Life or Tuya Mobile app?
2. I can then add it to Tuya-local Integration
3. After adding all wifi devices to Tuya-Local I will no longer need default HA Tuya integration and I can safely remove it.
4. Can i Then remove the device from the Smart Life/Tuya App or does it have to remain there.

**Comment by make-all** [comment]
1. Yes, the devices have to be paired first, they will not have a local key until they are paired.
2. After getting the local key and device key, you can add them to this integration.
3. You can move your devices over one by one, you can also leave them configured in the cloud integration until you are confident they are working correctly in this integration (this is only the case for moving from cloud to local - if you are moving from localtuya to tuya-local, the devices can only be used with one local integration at any time)
4. No, if you remove it from the app, the device may go back into pairing mode.  Just leave the app closed.  Some people have success blocking the devices from reaching the internet, but some devices seem to revert to pairing mode after some time not connecting to the cloud.

**Reply by yarafie** [reply]
Thanks a lot for the quick response.

Just spent an hour migrating from HA Tuya to Local-Tuya working flawlessly.

I do have a motion sensor which seems I will have to keep it in the cloud so have to keep HA Tuya. Until Inteplace it with s zigbee version probaly or figure a way to continuosly power it.
I assume I should just disable all other devices in HA Tuya that I moved to Tuya-local so they dont appear in my automations/alexa etv and only from Tuya-Local..

Last question, some of the switches have a default power on setting doesnt (at least in the app)  but i dont see that as an entity under Tuya-Local when added. Below is dump  from tinytuya, I redacted some information not sute if u need it.

Great Work again

{
        "name": "Clean Kitchen Switches",
        "id": "iiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiii",
        "key": "kkkkkkkkkkkkkkk",
        "mac": "mmmmmmmmmm",
        "uuid": "uuuuuuuuuuuuuuu",
        "sn": "EZNDF28FX01EB9",
        "category": "kg",
        "product_name": "WiFi Switch 2",
        "product_id": "bqaykzzru7fwzyzl",
        "biz_type": 18,
        "model": "\u4e8c\u8def\u5f00\u5173 CB3S",
        "sub": false,
        "icon": "https://images.tuyaeu.com/smart/icon/ay15422864509092y6k8/31ab>
        "ip": "[192.168.100.142](http://192.168.100.142/)",
        "version": "3.3"
    },


        {
            "id": "iiiiiiiiiiiiiiiiiiiiiiiiiiiiiiii",
            "ip": "[192.168.100.142](http://192.168.100.142/)",
            "active": 2,
            "ablilty": 0,
            "encrypt": true,
            "productKey": "key5ckdnys3pqgec",
            "name": "Clean Kitchen Switches",
            "key": "kkkkkkkkkkkkkkkkk",
            "mac": "mmmmmmmmmm",
            "token": "",
            "wf_cfg": "",
            "dev_type": "default",
            "err": "",
            "type": "default",
            "dps": {
                "dps": {
                    "1": true,
                    "2": true,
                    "7": 0,
                    "8": 0,
                    "14": "off",
                    "17": "",
                    "18": "",
                    "19": ""
                }
            },
            "origin": "broadcast",
            "ver": "3.3"
        },

        {
            "active_time": [1674907474](tel:1674907474),
            "biz_type": 18,
            "category": "kg",
            "create_time": [1674907474](tel:1674907474),
            "icon": "smart/icon/ay15422864509092y6k8/31ab10b01dc271ae608db90e7d>
            "id": "iiiiiiiiiiiiiiiiiiiiiiiiiiiiii",
            "ip": "[141.179.10.](http://141.179.10.17/)xx",
            "lat": "26.3100",
            "local_key": "kkkkkkkkkkkkkkkk",
            "lon": "50.1800",
            "model": "\u4e8c\u8def\u5f00\u5173 CB3S",
            "name": "Clean Kitchen Switches",
            "online": true,
            "owner_id": "77483093",
            "product_id": "bqaykzzru7fwzyzl",
            "product_name": "WiFi Switch 2",
            "status": [
                {
                    "code": "switch_1",
                    "value": true
                },
                {
                    "code": "switch_2",
                    "value": true
                },
                {
                    "code": "countdown_1",
                    "value": 0
                },
                {
                    "code": "countdown_2",
                    "value": 0
                },
                {
                    "code": "relay_status",
                    "value": "power_off"
                },
                {
                    "code": "cycle_time",
                    "value": ""
                },
                {
                    "code": "random_time",
                    "value": ""
                },
                {
                    "code": "switch_inching",
                    "value": ""
                }
            ],
            "sub": false,
            "time_zone": "+03:00",
            "uid": "uuuuuuuuuuuuuuuuu",
            "update_time": [1681983848](tel:1681983848),
            "uuid": "dddddddddddddddd"
        },
